Add 4th option -- "expand (no zoom)".

Vergir opened the pull request:
## Description

It's basically the same as "expand" but min_size argument in set_screen_stretch() is a constant equal to vanilla game (1920x1080), instead of being equal to viewport. This makes it so
- The UI size scales with bigger resolutions, like in vanilla. In current "expand", bigger resolutions make UI progressively smaller.
- The camera "zoom" is also same as vanilla, but the player will see more on the sides if their screen is extra wide/tall. In current "expand" bigger resolutions tend to move camera way too far away so that the arena doesn't even fill the screen.

A similar issue was solved in [this earlier commit](https://github.com/BrotatoMods/KANA-MultiRes/commit/d51bf008980635d27b1988cc93373b7fd0e84e16), where viewport adjusting made game uncomfortable for lower resolutions (Steam Deck). This Pull Request is a similar treatment for bigger resolutions (4K and more).

I also made several tiny changes mostly related to typos and codestyle, hope they're ok.

## Examples

Belor is what the game looks like on my display with the current "Expand" option. There's a huge border around the arena and the GUI is at times too small to read

![image](https://github.com/user-attachments/assets/b73bf509-a428-49cd-8ae0-da7c4558dec0)

Below is a new "expand (no zoom)" option. Much closer to vanilla experience

![image](https://github.com/user-attachments/assets/4c136619-4b09-4e5e-ba7d-677b2135f460)

For reference, [here's](https://github.com/user-attachments/assets/378149df-979e-4a32-a1c0-82493e8afb64) unmodded experience (also "default" setting in this mod)

Vergir committed:
Add 4th option: expand (no zoom)

- 4th option -- expand (no zoom) -- same as expand but without stretching to viewport size. This keeps UI size and Camera Zoom consistent at big resolutions, mostly noticeable on 4K and above.
- Small code refactorings here and there for better maintenance

Vergir committed:
Update missing settings button from Steam Workshop

Vergir commented on root/mods-unpacked/KANA-MultiRes/extensions/singletons/progress_data.gd:
```diff
@@ -2,8 +2,10 @@ extends "res://singletons/progress_data.gd"
-	"OPTION_SCREEN_STRETCH_EXPAND_WIDTH"]
+	"OPTION_SCREEN_STRETCH_EXPAND",
+	"OPTION_SCREEN_STRETCH_EXPAND_WIDTH",
+	"expand (no zoom)"
```
Unfortunately I couldn't provide a translation key here as the mod supplies a compiled .translation file that I can't easily modify. Hopefully you can fill it if you wish (feel free to rename the option as well).

KANAjetzt replied:
`"expand (no zoom)"` can we make that clearer (to be fair all option names are very ambiguous), but if I get it right it prevent's the zoom out right? So maybe "expand (no zoom out)" or "expand (keep zoom)"

Vergir replied:
I like "expand (no zoom out)", changed to that.

Vergir commented on root/mods-unpacked/KANA-MultiRes/extensions/singletons/utils.gd:
```diff
@@ -40,8 +40,21 @@ func set_strecht_mode() -> void:
 			SceneTree.STRETCH_ASPECT_KEEP,
 			Vector2(width, height)
 			)
+	# Settings set to "expand"
```
Mostly reordering of code blocks so they go 0 => 1 => 2 => (new) 3.

Vergir commented on root/mods-unpacked/KANA-MultiRes/manifest.json:
```diff
@@ -1,8 +1,8 @@
 	"namespace": "KANA",
-	"version_number": "2.3.1",
-	"description": "Allows you to set 3 different screen settings under the general settings tab. \n \nChangelog - v2.3.1\n [indent]- Fixed an issue where the stat popup appears behind the pause menu. [/indent]\n [indent]- Ty lx and Nodoka for reporting the issue![/indent]\n\nChangelog - v2.3.0\n [indent]- Updated for Brotato v1.1.0.1 [/indent]\n\nChangelog - v2.2.0\n [indent]- Updated for the new M ...
+	"version_number": "2.3.2",
```
This is a change I pulled from Steam Workshop mod files. I assume they didn't make it to github yet, so I put them here as a separate commit.

On a side note, I didn't update this description or README with info about new option. I think you'd want to do it yourself, but tell if you want me to fill anything.

KANAjetzt replied:
> On a side note, I didn't update this description or README with info about new option. I think you'd want to do it yourself, but tell if you want me to fill anything.

Please add a description for your option, you can describe it better then I can :)

Vergir replied:
Done 👍

KANAjetzt commented:
How does it compare to the "expand width" option? On my screens expand width and expand no zoom look the same.
I tested with:

- 2560x1440
- 3440x1440

Vergir commented:
> How does it compare to the "expand width" option? On my screens expand width and expand no zoom look the same. I tested with:
> 
>     * 2560x1440
> 
>     * 3440x1440

That's because you have only widescreen resolutions
On my screen (2880x2560) expand width looks like default
https://imgur.com/a/r3yG6rD

My understanding of options is:
- "default" is no stretching, no zoom out.
- "expand" is stretching, with zoom out
- "expand (no zoom out)" is stretching, no zoom out
- "expand width" is
\- for screens wider than 16:9 (your case): stretching, no zoom out, aka same as the "expand (no zoom out)"
\- for screens narrower than 16:9 (my case): no stretching, no zoom, aka same as default

Vergir committed:
rename no-zoom-out option, update readme

- 4th option is now "expand (no zoom out)", was "expand (no zoom)"
- Add description to README.MD and manifest.json

Vergir committed:
remove quotes from manifest.json update

Vergir committed:
Merge branch 'main' into main

Vergir commented:
Hey, thanks for taking time.

I updated the PR in accordance with your comments, and also added description to README.MD and manifest.json.

KANAjetzt commented on README.md:
```diff
@@ -34,3 +34,13 @@ Tested with Game Version: 1.1.4.0
+
+  ## expand (no zoom out)
+   
+   Scales width and height for non 16:9 resolutions, but unlike "expand" it keeps zoomed in look and feel of vanilla game. 
```
```suggestion
   Scales width and height for non 16:9 resolutions, but unlike "expand" it keeps the zoomed in look and feel of the vanilla game. 
```

Vergir replied:
Applied it in another commit, where I replaced imgur links.
Thanks for the catch!

KANAjetzt commented on README.md:
```diff
@@ -34,3 +34,13 @@ Tested with Game Version: 1.1.4.0
+   
+   Scales width and height for non 16:9 resolutions, but unlike "expand" it keeps zoomed in look and feel of vanilla game. 
+   <br />
+   Examples album for 9:8 (2880 x 2560) resolution: https://imgur.com/a/2880x2560-screen-options-r3yG6rD
```
you can upload the screenshots directly to Github by drag and dropping them into the text editor. 
You can use /details to hide them behind a details block 👍

Vergir replied:
Done 👍

KANAjetzt commented on root/mods-unpacked/KANA-MultiRes/extensions/main.gd:
```diff
@@ -62,4 +62,4 @@ func KANA_add_ARC():
 	KANA_change_background()
 	KANA_add_ARC()
-	Utils.set_strecht_mode()
+	Utils.set_stretch_mode()
```
thanks for fixing the typos 😄

KANAjetzt commented on root/mods-unpacked/KANA-MultiRes/extensions/singletons/progress_data.gd:
```diff
@@ -2,8 +2,10 @@ extends "res://singletons/progress_data.gd"
-	"OPTION_SCREEN_STRETCH_EXPAND_WIDTH"]
+	"OPTION_SCREEN_STRETCH_EXPAND",
+	"OPTION_SCREEN_STRETCH_EXPAND_WIDTH",
+	"expand (no zoom out)"
```
I add this to the translations file after merge, and add translations csv to the repo 👍

KANAjetzt reviewed (changes requested):
Nice thanks again, the only thing is the img link to imgur.

Vergir committed:
Update README.md, replace imgur links with github ones

Vergir commented:
Updated!
